addons/mass_mailing/models/mailing.py

```diff
@@ -525,6 +527,51 @@ def _get_remaining_recipients(self):
         done_res_ids = [record['res_id'] for record in already_mailed]
         return [rid for rid in res_ids if rid not in done_res_ids]
 
+    def _send_statistics_mails(self):
+        """Send an email to the responsible of each finished mailing with the statistics."""
+        self.kpi_mail_required = False
+
+        for mailing in self:
+            user = mailing.user_id
+
+            link_trackers = self.env['link.tracker'].search(
+                [('mass_mailing_id', '=', mailing.id)]
+            ).sorted('count', reverse=True)
+
+            link_trackers_body = self.env['mail.render.mixin']._render_encapsulate(
```

Comment:
Encapsulate is meant to add an envelope around existing html code. Here a simple render is sufficient isn't it ? Or use _render_template_qweb .

Reply:
Indeed :)